codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -326,17 +326,26 @@ async def _handle_execution(
                                     )
                                 except SmithyRetryException:
                                     raise context_with_response.response
+                                logger.debug(
+                                    "Retry needed. Attempting request #%s in %s seconds.",
```

Comment:
Does retry_count start at 0 or 1 on our first call? i.e. is the first log message request 1 or 2?

Reply:
Ok if the first log starts at 2, then this probably fine. I would say let's constrain the formatting a little though for seconds. Something like `%.3f` (or 4) should work to give us enough relevant information. I don't think we need full precision for this.

Reply:
I went with `%.4f`. This was added in 36b28c71fa60e109aa90b1bf5088509a8ecd6ca6